### PR TITLE
feat(university-gateway): Frontend changes (use of query params, translations, pagination smoothed, minor UI/UX)

### PR DESCRIPTION
**Pull request by Ballioli**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs
Pasted Graphic 14
Application period is now correctly displayed on 'en' locale

Umhverfisbreytingar á
New attributes added that are received from UGLA/HR(later) that determine if programme can be applied to and if the application is available on island.is(link to island.is application will be added later when application testing is done)

University of Iceland
English translations for university names on cards

https://github.com/island-is/island.is/assets/22780994/94ea215d-5e4c-454b-89a9-de0caa0422c0
Initial filter no longer shifts cards down creating a shaking motion

https://github.com/island-is/island.is/assets/22780994/5392dcf5-52d1-498f-a1cb-c6d883fb98eb
Search string and filters are now saved in query parameters instead of locale storage.

https://github.com/island-is/island.is/assets/22780994/fa75ae58-a3a0-4647-8f86-60c45adff49d
Pagination now scrolls the user to the title of the page

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
    - Added properties to track open application periods and university-specific application statuses in program listings.
    - Enhanced university search functionality with new filter handling and application status displays.
    - Updated navigation links in the university studies header for better user redirection.

- **Refactor**
    - Improved button and link interaction within various components for a more intuitive user experience.
    - Streamlined rendering logic in the university search details screen, including locale-specific date formatting.

- **Bug Fixes**
    - Fixed the disabled state of buttons based on application availability in university search details.

- **Documentation**
    - Updated internal documentation for new and modified properties and methods related to university applications.

### CODE REVIEW

**Review by berglindoma13 — APPROVED** [review]
LGTM

File: apps/web/screens/UniversitySearch/UniversitySearch.tsx [hunk 1]
```diff
@@ -734,9 +814,7 @@ const UniversitySearch: Screen<UniversitySearchProps> = ({
                       variant="text"
                       icon="reload"
                       size="small"
-                      onClick={() =>
-                        setFilters(JSON.parse(JSON.stringify(initialFilters)))
-                      }
+                      onClick={() => clearFilterParams()}
```
Comment (RunarVestmann) [inline apps/web/screens/UniversitySearch/UniversitySearch.tsx:817]: Probably enough to just do

onClick={clearFilterParams}

File: apps/web/screens/UniversitySearch/UniversitySearch.tsx [hunk 2]
```diff
@@ -576,6 +619,43 @@ const UniversitySearch: Screen<UniversitySearchProps> = ({
     }
   }
 
+  const handleUserInput = (value: string) => {
+    setSelectedPage(1)
+    searchTermHasBeenInitialized.current = true
+    setQuery(value)
+
+    const currentQueryParams = router.query
+    const updatedQueryParams = {
```
Comment (RunarVestmann) [inline apps/web/screens/UniversitySearch/UniversitySearch.tsx:628]: Perhaps check here if the value is empty then we might want to just stop having the search param in the url.

This is handled by the usequerystate library which I've wanted to try out more: https://www.npmjs.com/package/next-usequerystate

It's already in the package.json if you want to try it out

File: apps/web/components/ActionCategoryCard/ActionCategoryCard.tsx [hunk 3]
```diff
@@ -141,22 +141,23 @@ const Component = forwardRef<HTMLElement, ActionCategoryCardProps>(
             display="flex"
             justifyContent={['flexStart', 'flexEnd']}
             flexDirection="row"
+            style={{ cursor: cta.disabled ? 'not-allowed' : '' }}
```
Comment (RunarVestmann) [inline apps/web/components/ActionCategoryCard/ActionCategoryCard.tsx:144]: It's probably harmless to set the cursor to an empty string but maybe it's safer to set it to undefined so it behaves the default way

File: apps/web/components/ListViewCard/ListViewCard.tsx [hunk 4]
```diff
@@ -125,21 +125,27 @@ export const ListViewCard = ({
           </Box>
         )}
         <Box paddingTop={3} width="full" height="full">
-          <Box display={'flex'} alignItems={'flexEnd'} height="full">
+          <Box
+            display={'flex'}
+            alignItems={'flexEnd'}
+            height="full"
+            width="full"
+            style={{ cursor: cta.disabled ? 'not-allowed' : '' }}
```
Comment (RunarVestmann) [inline apps/web/components/ListViewCard/ListViewCard.tsx:133]: It's probably harmless to set the cursor to an empty string but maybe it's safer to set it to undefined so it behaves the default way

File: apps/web/components/ListViewCard/ListViewCard.tsx [hunk 5]
```diff
@@ -125,21 +125,27 @@ export const ListViewCard = ({
           </Box>
         )}
         <Box paddingTop={3} width="full" height="full">
-          <Box display={'flex'} alignItems={'flexEnd'} height="full">
+          <Box
+            display={'flex'}
```
Comment (RunarVestmann) [inline apps/web/components/ListViewCard/ListViewCard.tsx:129]: display="flex" is 2 fewer characters (nitpick, doesn't really matter, do what you want with this information)

File: apps/web/components/ListViewCard/ListViewCard.tsx (hunk 4 above)
Comment (Ballioli) [reply apps/web/components/ListViewCard/ListViewCard.tsx:133]: Agree, fixed

File: apps/web/components/ActionCategoryCard/ActionCategoryCard.tsx (hunk 3 above)
Comment (Ballioli) [reply apps/web/components/ActionCategoryCard/ActionCategoryCard.tsx:144]: Agree, fixed

File: apps/web/screens/UniversitySearch/UniversitySearch.tsx (hunk 2 above)
Comment (Ballioli) [reply apps/web/screens/UniversitySearch/UniversitySearch.tsx:628]: Updated to remove empty search param. Will check the library another time.

File: apps/web/screens/UniversitySearch/UniversitySearch.tsx (hunk 1 above)
Comment (Ballioli) [reply apps/web/screens/UniversitySearch/UniversitySearch.tsx:817]: Fixed

**Pull request merged**